cdap-api/src/main/java/co/cask/cdap/api/retry/RetryPolicy.java

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.retry;
+
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+
+/**
+ * Defines retry behavior when CDAP encounters an error during a retry-able operation.
+ * Users should use the {@link #builder()} to create a RetryPolicy.
+ */
+public class RetryPolicy {
```

Comment:
I wonder whether this could be an interface, with different implementations? And then you configure the class name plus a set of properties? That would give more flexibility for future improvements/variations.